dev-python/pypdf/pypdf-4.3.1.recipe

```diff
@@ -0,0 +1,69 @@
+SUMMARY="A pure-python PDF library"
+DESCRIPTION="pypdf is a free and open-source pure-python PDF library capable of splitting, \
+merging, cropping, and transforming the pages of PDF files. It can also add custom data, viewing \
+options, and passwords to PDF files. pypdf can retrieve text and metadata from PDFs as well."
+HOMEPAGE="https://pypdf.readthedocs.io/"
+COPYRIGHT="2006-2008, Mathieu Fenniak
+	2007, Ashish Kulkarni
+	2014, Steve Witham"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+pypiVersion="3c/60/eccdd92dd4af3e4bea6d6a342f7588c618a15b9bec4b968af581e498bcc4"
+SOURCE_URI="https://files.pythonhosted.org/packages/$pypiVersion/pypdf-$portVersion-py3-none-any.whl#noarchive"
+CHECKSUM_SHA256="64b31da97eda0771ef22edb1bfecd5deee4b72c3d1736b7df2689805076d6418"
+
+ARCHITECTURES="any"
+DISABLE_SOURCE_PACKAGE="yes"
+
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PYTHON_VERSIONS=(3.10)
+for i in "${!PYTHON_VERSIONS[@]}"; do
+	pythonVersion=${PYTHON_VERSIONS[$i]}
+	pythonPackage=python${pythonVersion//.}
+	eval "PROVIDES_$pythonPackage=\"
+		${portName}_$pythonPackage = $portVersion
+		\"
+	REQUIRES_$pythonPackage=\"
+		haiku
+		cmd:python$pythonVersion
```

Comment:
This has some optional dependencies:
 * cryptography for supporting encrypted PDFs
 * Pillow for extracting images

I suppose I should leave them out and not require them? @OscarL

Reply:
Thanks, then I'll leave them out. The old version of psutils didn't even support PDF at all, so that is already an improvement.

I guess that this is needed for supporting PDF. For psutils, it is a hard requirement.